main/src/library/Foldable.flix

```diff
@@ -80,24 +80,24 @@ pub class Foldable[t : Type -> Type] {
     ///
     /// Returns `false` if `t` is empty.
     ///
-    pub def exists(f: a -> Bool, t: t[a]): Bool =
-        Foldable.foldLeft((acc, x) -> acc or f(x), false, t)
+    pub def exists(f: a -> Bool, t: t[a]): Bool & ef =
+        Foldable.foldRightWithCont((x, k) -> k() or f(x), false, t)
 
     ///
     /// Returns `true` if and only if all elements in `t` satisfy the predicate `f`.
     ///
     /// Returns `true` if `t` is empty.
     ///
     pub def forall(f: a -> Bool, t: t[a]): Bool =
-        Foldable.foldLeft((acc, x) -> acc and f(x), true, t)
+        Foldable.foldRightWithCont((x, k) -> k() and f(x), true, t)
```

Comment:
This should be `f(x) and k()` this way if `f(x)` is false we dont evaluate the entire foldable. Makes sense?

Reply:
Same comment applies to exists.